lib/filewatch/watch.rb

```diff
@@ -126,77 +124,136 @@ def inode(path,stat)
     #   :create - file is created (new file after initial globs, start at 0)
     #   :modify - file is modified (size increases)
     #   :delete - file is deleted
-    public
     def each(&block)
       synchronized do
+        return if @files.empty?
+
+        file_deleteable = []
+        # look at the closed to see if its changed
+
+        @files.values.select {|wf| wf.closed? }.each do |watched_file|
```

Comment:
> Hash#select returns an enumerable

It does and over a new smaller Hash that the `each` iterates over. What I really want is `each_selected_value(proc, block)` :smile: